### PR TITLE
QC migration patches

### PR DESCRIPTION
**Pull request by froggleston**
Fix a few QC issues when migrating.

### CODE REVIEW

File: sqlstore/src/main/resources/db/migration/V0320__auto_main.sql [hunk 1]
```diff
@@ -45,11 +44,11 @@ CREATE TABLE `PoolQC2` (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
 INSERT INTO LibraryQC2(`qcId`, `library_libraryId`, `creator`, `date`, `type`, `results`)
-	SELECT qcId, library_libraryId, (SELECT userId FROM `User` WHERE fullName = qcCreator), qcDate, qcMethod, results FROM LibraryQC;
+	SELECT qcId, library_libraryId, IFNULL((SELECT userId FROM `User` WHERE fullName = qcCreator OR loginName = qcCreator),1), qcDate, qcMethod, results FROM LibraryQC;
```
Comment (apmasell) [inline sqlstore/src/main/resources/db/migration/V0320__auto_main.sql]: 1. I don't think a substitute user is appropriate. If your database has bad data, it should fail to migrate.
2. Why user 1? It's not even guaranteed to be admin. I believe it isn't admin in your database.

File: sqlstore/src/main/resources/db/migration/V0320__auto_main.sql [hunk 2]
```diff
@@ -81,7 +80,7 @@ CREATE TABLE PlatformSizes (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
 INSERT INTO PlatformSizes(platform_platformId, partitionSize)
-  SELECT DISTINCT platform, COUNT(*) AS c FROM SequencerPartitionContainer JOIN SequencerPartitionContainer_Partition ON containerId = container_containerId GROUP BY containerId
+  SELECT platform, c FROM (SELECT DISTINCT platform, COUNT(*) AS c FROM SequencerPartitionContainer JOIN SequencerPartitionContainer_Partition ON containerId = container_containerId GROUP BY containerId) as p where platform > 0
```
Comment (apmasell) [inline sqlstore/src/main/resources/db/migration/V0320__auto_main.sql]: If the platform is 0, the database is wrong and should be fixed. How is this possible? You have containers with no associated platform? Are they attached to a sequencer?

File: sqlstore/src/main/resources/db/migration/V0320__auto_main.sql (hunk 1 above)
Comment (froggleston) [reply sqlstore/src/main/resources/db/migration/V0320__auto_main.sql]: I'm unsure how to represent this in this instance. We can't delete the QCs. We could map them to an existing user in our DB, but then we'll lose the real data. The only solution we have if we can't map it to an existing user is to create a new user for each external QC creator. I'll speak to the team here.

File: sqlstore/src/main/resources/db/migration/V0320__auto_main.sql (hunk 2 above)
Comment (froggleston) [reply sqlstore/src/main/resources/db/migration/V0320__auto_main.sql]: I've checked this and these containers can be deleted as they're duplicates.

**Comment by froggleston** [comment]
This can be closed now. I've worked through the local DB issues so that the develop migrations now work properly.

**Pull request closed without merging**